lib/Jsrt/JsrtDebugManager.cpp

```diff
@@ -0,0 +1,693 @@
+                {
+                    Adelete(jsrtDebugManager->GetDebugObjectArena(), jsrtDebugManager->stackFrames);
+                    jsrtDebugManager->stackFrames = nullptr;
+                }
+            }
+
+            this->jsrtDebugManager = nullptr;
+        }
+    private:
+        JsrtDebugManager* jsrtDebugManager;
+    };
+
+    auto funcPtr = [&]()
+    {
+        AutoClear autoClear(this);
+        this->debugEventCallback(debugEvent, eventDataObject, this->callbackState);
+    };
+
+    if (scriptContext->GetThreadContext()->IsScriptActive())
+    {
+        BEGIN_LEAVE_SCRIPT(scriptContext)
+        {
+            funcPtr();
+        }
+        END_LEAVE_SCRIPT(scriptContext);
+    }
+    else
+    {
+        funcPtr();
+    }
+}
+
+void JsrtDebugManager::CallDebugEventCallbackForBreak(JsDiagDebugEvent debugEvent, Js::DynamicObject* eventDataObject, Js::ScriptContext* scriptContext)
+{
+    AutoSetDispatchHaltFlag autoSetDispatchHaltFlag(scriptContext, scriptContext->GetThreadContext());
+
+#if DBG
+    void *frameAddress = _AddressOfReturnAddress();
+    scriptContext->GetThreadContext()->GetDebugManager()->SetDispatchHaltFrameAddress(frameAddress);
+#endif
+
+    this->CallDebugEventCallback(debugEvent, eventDataObject, scriptContext);
+
+    for (Js::ScriptContext *tempScriptContext = scriptContext->GetThreadContext()->GetScriptContextList();
+    tempScriptContext != nullptr && !tempScriptContext->IsClosed();
+        tempScriptContext = tempScriptContext->next)
+    {
+        tempScriptContext->GetDebugContext()->GetProbeContainer()->AsyncDeactivate();
+    }
+
+    if (Js::Configuration::Global.EnableJitInDebugMode())
+    {
+        scriptContext->GetThreadContext()->GetDebugManager()->GetDebuggingFlags()->SetForceInterpreter(false);
+    }
+}
+
+Js::DynamicObject* JsrtDebugManager::GetScript(Js::Utf8SourceInfo* utf8SourceInfo)
+{
+    Js::DynamicObject* scriptObject = utf8SourceInfo->GetScriptContext()->GetLibrary()->CreateObject();
+
+    JsrtDebugUtils::AddScriptIdToObject(scriptObject, utf8SourceInfo);
+    JsrtDebugUtils::AddFileNameOrScriptTypeToObject(scriptObject, utf8SourceInfo);
+    JsrtDebugUtils::AddLineCountToObject(scriptObject, utf8SourceInfo);
+    JsrtDebugUtils::AddPropertyToObject(scriptObject, JsrtDebugPropertyId::sourceLength, utf8SourceInfo->GetCchLength(), utf8SourceInfo->GetScriptContext());
+
+    return scriptObject;
+}
+
+Js::JavascriptArray* JsrtDebugManager::GetScripts(Js::ScriptContext* scriptContext)
+{
+    Js::JavascriptArray* scriptsArray = scriptContext->GetLibrary()->CreateArray();
+
+    int index = 0;
+
+    for (Js::ScriptContext *tempScriptContext = scriptContext->GetThreadContext()->GetScriptContextList();
+    tempScriptContext != nullptr && !tempScriptContext->IsClosed();
+        tempScriptContext = tempScriptContext->next)
+    {
+        tempScriptContext->GetSourceList()->Map([&](int i, RecyclerWeakReference<Js::Utf8SourceInfo>* utf8SourceInfoWeakRef)
```

Comment:
changed at all places

---
In reply to: [62760284](https://github.com/Microsoft/ChakraCore/pull/926#discussion_r62760284) [](ancestors = 62760284)